tests/test_md.py

```diff
@@ -640,6 +692,127 @@ def test_multiple_steps_local_no_free_particles(self):
+
+        intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+
+        steps = 10
+
+        # Construct context with no potentials, should fail to initialize.
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), [])
+        with pytest.raises(RuntimeError, match="unable to find a NonbondedAllPairs potential"):
+            ctxt.initialize_local_md()
+
+        # If you have multiple nonbonded potentials, should fail
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps * 2)
+        with pytest.raises(RuntimeError, match="found multiple NonbondedAllPairs potentials"):
+            ctxt.initialize_local_md()
+
+        # Verify that initializing local md doesn't modify global md behavior
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps)
+        ref_xs, ref_boxes = ctxt.multiple_steps(steps)
+
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps)
+        ctxt.initialize_local_md()
+        comp_xs, comp_boxes = ctxt.multiple_steps(steps)
+
+        np.testing.assert_array_equal(ref_xs, comp_xs)
+        np.testing.assert_array_equal(ref_boxes, comp_boxes)
+
+        # Verify that initializing local md doesn't modify global md behavior
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps)
+        ref_local_xs, ref_local_boxes = ctxt.multiple_steps_local(steps, local_idxs)
+
+        ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps)
+        ctxt.initialize_local_md()
+        comp_local_xs, comp_local_boxes = ctxt.multiple_steps_local(steps, local_idxs)
+
+        np.testing.assert_array_equal(ref_local_xs, comp_local_xs)
+        np.testing.assert_array_equal(ref_local_boxes, comp_local_boxes)
+
+    def test_local_md_with_selection_mask(self):
+        """Verify that running local md with a selection mask works as expected"""
+        seed = 2023
+        np.random.seed(seed)
+        rng = np.random.default_rng(seed)
+
+        N = 8
+        D = 3
+
+        coords = rng.random(size=(N, D)).astype(dtype=np.float64)
+        box = np.eye(3) * 3.0
+        masses = rng.random(N)
+
+        E = 2
+
+        params, potential = prepare_nb_system(
+            coords,
+            E,
+            p_scale=3.0,
+            cutoff=1.0,
+        )
+        nb_pot = potential.to_gpu(np.float32)
+
+        temperature = 300
+        dt = 1.5e-3
+        friction = 0.0
+
+        v0 = np.zeros_like(coords)
+        bps = [nb_pot.bind(params).bound_impl]
+
+        intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+
+        idxs = np.arange(0, len(coords))
+
+        reference_idx = rng.choice(idxs)
+
+        free_particles = rng.choice(idxs, size=len(coords) // 2, replace=False)
+        free_particles = np.delete(free_particles, free_particles == reference_idx)
+        frozen_particles = np.delete(idxs, free_particles)
+
+        steps = 10
+
+        # Verify that initializing local md doesn't modify global md behavior
```

Comment:
```suggestion
        # Verify that initializing local md doesn't modify local md behavior
```

Reply:
This line is copy-pasta, should be removed